libraries/AP_L1_Control/AP_L1_Control.cpp

```diff
@@ -123,8 +123,28 @@ int32_t AP_L1_Control::target_bearing_cd(void) const
  */
 float AP_L1_Control::turn_distance(float wp_radius) const
 {
-    wp_radius *= sq(_ahrs.get_EAS2TAS());
-    return MIN(wp_radius, _L1_dist);
+    float eas2tas_sq = sq(_ahrs.get_EAS2TAS()); // true airspeed correction
+
+    float scaled_turn_distance = wp_radius * eas2tas_sq; // distance scaled to true airspeed
+
+    /** Wind effect i.e. forward crosstrack error correction for the leg to the next waypoint.
+     *  turn early on tail-wind
+     */
+    float headwind = _ahrs.head_wind(); // forward head-wind component in m/s. Negative means tail-wind.
```

Comment:
You want wind component in the direction of the current leg. Not at the current yaw angle.

Reply:
The simplification of turning 90 degrees makes current tailwind the crosswind after the turn.
No additional code and calculations necessary because AHRS already has done it.
L1_Control scales other turns based on the 90 degree turn, so this correction turns up in every turn angle.